### DISCUSSION TITLE
Guidance on creating links between documents when hosting on readthedocs

### DISCUSSION THREAD

**Discussion by Midnighter** [mkdocs/mkdocs#3230] [Q&A] [answered]
I have some trouble with absolute links and linking to anchors within markdown documents when hosting the documentation on readthedocs.

The first problem is that readthedocs will insert a path into the URL, for example, https://project.readthedocs.io becomes https://project.readthedocs.io/en/latest. This breaks any absolute links of the form `[go home](/home)` which will point to https://project.readthedocs.io/home/ which doesn't exist. I believe, this could be a problem with my [site_url](https://www.mkdocs.org/user-guide/configuration/#site_url), which I have set to https://project.readthedocs.io/ but should be variable since I'm serving multiple versions on readthedocs.

My second problem is, how do I create an anchor link to an `index.md`?  When I use an absolute link to the directory, for example, `[about](/#about)` I run into the first problem again. When I use a relative link, I need to "go up an extra directory" which is unintuitive but probably due to the default [use_directory_urls: true](https://www.mkdocs.org/user-guide/configuration/#use_directory_urls).

Can anyone provide me with best practices/guidance how to best avoid these issues? Thanks in advance.

**Comment by oprypin** [comment, marked as answer]
You need to use only relative links  and link only to Markdown files or other real files in the docs dir. Such as `about.md`, `../about.md`, `../about/index.md`. Not to their imaginary destinations like `about.html`, `about/index.html`, `about/`, `/about` or anything of the sort. Think of it like this: if you put the files as is onto GitHub, its preview will also cross-link everything correctly.

**Reply by gamesbook** [reply]
This does not work for RestructuredText; links such <about.rst> which work perfectly in GitHub, fail on the  readthedocs pages; where it seems to need specific <about.html> links.

**Reply by pawamoy** [reply]
MkDocs uses Markdown, not reStructuredText 🫤

**Reply by gamesbook** [reply]
I appreciate that; I was just noting the difference - it seems a bit strange that one markup format works but not the other?

**Reply by pawamoy** [reply]
Oh, my bad! Yeah rST is probably less supported by GitHub :shrug:
